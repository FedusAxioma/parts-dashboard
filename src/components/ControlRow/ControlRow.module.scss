.row {
	height: 40px;
	width: 100%;
	display: flex;
	align-items: center;

	span:nth-of-type(1) {
		width: 30%;
	}
	span:nth-of-type(2) {
		width: 20%;
		text-align: center;
	}
	span:nth-of-type(3) {
		width: 30%;
		text-align: center;
	}
	span:nth-of-type(4) {
		width: 20%;
	}
}
