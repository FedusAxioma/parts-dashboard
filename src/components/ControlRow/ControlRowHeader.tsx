import ControlRow from './ControlRow';
import React from 'react';

const ControlRowHeader: React.FC = () => {
	return (
		<ControlRow name={'Control'} deviation={'Dev'} deviationOutTotal={'Dev Out Tol'} />
	);
};

export default ControlRowHeader;
