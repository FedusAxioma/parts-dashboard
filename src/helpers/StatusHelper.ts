import { AiFillCheckCircle, AiFillCloseCircle, AiFillWarning } from 'react-icons/ai';
import { Status, StatusValue } from '../interfaces/Status';

const getStatusValues = (status: Status): StatusValue => {
	switch (status) {
		case 'GOOD':
			return {
				color: 'green',
				icon: AiFillCheckCircle,
			};
		case 'WARNING':
			return {
				color: '#e2e201',
				icon: AiFillWarning,
			};
		case 'ERROR':
		default:
			return {
				color: 'red',
				icon: AiFillCloseCircle,
			};
	}
};

export default {
	getStatusValues,
};
