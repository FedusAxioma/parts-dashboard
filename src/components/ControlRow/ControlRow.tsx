import React from 'react';
import { Status } from '../../interfaces/Status';
import StatusHelper from '../../helpers/StatusHelper';
import styles from './ControlRow.module.scss';

interface ControlRowProps {
	name: string;
	deviation: number | string;
	deviationOutTotal: number | string;
	status?: Status;
}

const getStatusIcon = (currentStatus: Status): JSX.Element => {
	const statusValue = StatusHelper.getStatusValues(currentStatus);

	return <statusValue.icon style={{ color: statusValue.color }} />;
};

const ControlRow: React.FC<ControlRowProps> = ({ name, deviation, deviationOutTotal, status }) => {
	return (
		<div className={styles.row}>
			<span>{name}</span>
			<span>{deviation}</span>
			<span>{deviationOutTotal}</span>
			<span>
				{
					status && getStatusIcon(status)
				}
			</span>
		</div>
	);
};

export default React.memo(ControlRow);
