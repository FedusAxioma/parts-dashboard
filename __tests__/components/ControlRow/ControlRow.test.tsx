import { AiFillWarning } from 'react-icons/ai';
import ControlRow from '../../../src/components/ControlRow';
import { createRenderer } from 'react-test-renderer/shallow';
import React from 'react';
import StatusHelper from '../../../src/helpers/StatusHelper';

describe('ControlRow', () => {
	const shallow = createRenderer();

	test('renders component with all mandatory fields', () => {
		shallow.render(<ControlRow name="test" deviation={10} deviationOutTotal={20} />);

		expect(shallow.getRenderOutput()).toMatchSnapshot();
	});

	test('renders component with all fields', () => {
		StatusHelper.getStatusValues = jest.fn().mockReturnValue({
			color: 'blue',
			icon: AiFillWarning,
		});
		shallow.render(<ControlRow name="test" deviation={10} deviationOutTotal={20} status={'WARNING'} />);

		expect(shallow.getRenderOutput()).toMatchSnapshot();
	});
});
