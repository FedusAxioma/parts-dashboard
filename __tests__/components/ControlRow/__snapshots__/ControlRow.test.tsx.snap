// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`App renders component with all fields 1`] = `
<div>
  <span>
    test
  </span>
  <span>
    10
  </span>
  <span>
    20
  </span>
  <span>
    <AiFillWarning
      style={
        Object {
          "color": "blue",
        }
      }
    />
  </span>
</div>
`;

exports[`App renders component with all mandatory fields 1`] = `
<div>
  <span>
    test
  </span>
  <span>
    10
  </span>
  <span>
    20
  </span>
  <span />
</div>
`;
